src/borg/compress.pyx

```diff
@@ -306,44 +381,40 @@ class Auto(CompressorBase):
 
     def _decide(self, data):
         """
-        Decides what to do with *data*. Returns (compressor, lz4_data).
+        Decides what to do with *data*. Returns (compressor, compressed_data).
 
-        *lz4_data* is the LZ4 result if *compressor* is LZ4 as well, otherwise it is None.
+        *compressed_data* is the LZ4 result unless *compressor* is NONE, in which
+        case it is the uncompressed NONE compressor result.
         """
-        lz4_data = LZ4_COMPRESSOR.compress(data)
-        # lz4_data includes the compression type header, while data does not yet
-        ratio = len(lz4_data) / (len(data) + 2)
+        compressor, compressed_data = LZ4_COMPRESSOR.decide_compress(data)
+        # compressed_data includes the compression type header, while data does not yet
+        ratio = len(compressed_data) / (len(data) + 2)
         if ratio < 0.97:
-            return self.compressor, lz4_data
-        elif ratio < 1:
-            return LZ4_COMPRESSOR, lz4_data
+            return self.compressor, compressed_data
         else:
-            return NONE_COMPRESSOR, None
+            return compressor, compressed_data
 
     def decide(self, data):
         return self._decide(data)[0]
 
     def compress(self, data):
-        compressor, lz4_data = self._decide(data)
-        if compressor is LZ4_COMPRESSOR:
+        compressor, chp_compressed_data = self._decide(data)
```

Comment:
maybe write cheap if you mean cheap. :)

Reply:
I used chp_ solely to mathch the existing exp_ of the expensive one, I'll change both to the long form then. ;)

Reply:
oh, ok :)